src/components/NodeConnectionView.js

```diff
@@ -49,6 +51,25 @@ export default class NodeConnectionView extends Component<Props, State> {
         props.connectionError &&
         props.connectionError.startsWith('You need to stake'),
     }
+    this.resolveEns()
+  }
+
+  async resolveEns () {
+    const { ethNetwork } = this.props
+    const resolverContract = ResolverContract(ethNetwork)
+    const stakeContract = StakeContract(ethNetwork)
+    try {
+      const requiredStake = await stakeContract.requiredStake()
+      const stakeAddress = await resolverContract.resolve(ENS_NAMES.stake[ethNetwork])
+      const tokenAddress = await resolverContract.resolve(ENS_NAMES.token[ethNetwork])
+      this.setState({
+        stakeAddress,
+        tokenAddress,
+        requiredStake,
+      })
+    } catch (err) {
+      console.warn('err: ', err)
```

Comment:
We probably need to notify the user here, or at least prevent from trying to open MEW because we won't have the data to do it, unless it is handled somewhere else?

Reply:
yea good call, will add some feedback